Sources/BaggageLogging/Baggage+Logger.swift

```diff
@@ -0,0 +1,12 @@
+import Baggage
+import Logging
+
+extension BaggageContext {
+    public var logger: Logger {
+        var logger = Logger(label: "BaggageContext")
```

Comment:
I "fear" we may need to merry a few concepts here, specifically:

- some applications like to "create root logger and always pass it around and never use the global factory, and
- some applications don't care, and they'd be ok with us making up a logger like that.

What if we indeed try to allow the 1st case as well... Granted this is somewhat exploratory territory but let's give it a shot. 

It would indeed start moving us towards "context is a bag of random stuff" which I'm worried about but the upside is better composition and this pattern as well...

Say, what if we allow:

```suggestion
        var logger = self[BaseLoggerKey.self] ?? Logger(label: "BaggageContext")
```

this way a framework could still adopt this pattern...

Not sure if either style will be "the style" that will be adopted in the end, but we should explore these.

Let's also here on the documentation explain how it would be used and what it does.

With regards to copying all the metadata into the logger eagerly... I think that's where a custom logging handler could play a role; there is the `    internal init(label: String, _ handler: LogHandler)` overload which would allow us to implement the "materialize (make a string from it) values from context _only_ when we actually log things". Worth exploring!

Reply:
> It would indeed start moving us towards "context is a bag of random stuff"

This is really interesting. I didn't think about using the fact that we have "storage" at hand to "inject" a logger, pretty smart. However, as you already mentioned, we are somewhat crossing a line with this. But let's explore this further 👍

> I think that's where a custom logging handler could play a role

Interesting as well. I'll look into this too.